src/lib/formatter/SarifFormatter.ts

```diff
@@ -42,16 +43,16 @@ abstract class SarifFormatter {
 	 * Converts the rule violation to either "warning" or "error"
 	 * See https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html#_Toc10127839
 	 */
-	protected abstract getLevel(violation: RuleViolation): string;
+	protected abstract getLevel(violation: RuleViolation): Notification.level;
 
 	/**
 	 * Return a location that is used by violations and errors
 	 */
-	private getLocation(r: RuleResult): unknown {
+	private static getLocation(r: RuleResult): Location {
 		return {
 			physicalLocation: {
 				artifactLocation: {
-					uri: url.pathToFileURL(r.fileName)
+					uri: url.pathToFileURL(r.fileName).toString()
```

Comment:
`url.pathToFileURL` technically returns a URL, which the linter was complaining about. So I added the `.toString()` call.